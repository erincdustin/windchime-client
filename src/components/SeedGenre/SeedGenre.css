.genre-btn {
  margin-right: 2.5px;
  /* background-color: rgb(220, 235, 240); */
  /* border: .5px solid lightgrey; */
  /* border-radius: 10px; */
  padding: 3px;
}

.center {
  display: block;
  text-align: center;
}

.inline {
  display: inline-block;
  margin: 0px 10px;
}

.mapped-genres {
  font-family: 'Helvetica', 'Arial', sans-serif;
  text-align: center;
  background-color: rgb(228, 239, 240);
  border: 1px rgb(211, 211, 211) solid;
  border-radius: 10px;
  padding: 20px 0px;
}

.top-artists {
  background-color: rgb(228, 239, 240);
  border: 1px rgb(211, 211, 211) solid;
  border-radius: 10px;
  padding: 20px 0px;
}
