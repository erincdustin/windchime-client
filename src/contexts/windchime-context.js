import React, { Component } from 'react'

const WindChimeContext = React.createContext({
    error: '',
    songs: null,
    weather: null,
    genreChoice: null,
    id: null,
    artists: null,
    topArtists: null,
    playlistId: null,
    userPlaylists: null,
    snapshot: null,
    targetEnergy: null,
    targetValence: null,
    targetTempo: null,
    targetPopularity: null,
    setError: () => { },
    clearError: () => { },
    setSongs: () => { },
    setWeather: () => { },
    setGenreChoice: () => { },
    setId: () => { },
    setArtists: () => { },
    setTopArtists: () => { },
    setPlaylistId: () => { },
    setUserPlaylists: () => { },
    setSnapshot: () => { },
    setTargetEnergy: () => { },
    setTargetValence: () => { },
    setTargetTempo: () => { },
    setTargetPopularity: () => { },
})

export default WindChimeContext;

export class WindChimeProvider extends Component {
    constructor(props) {
        super(props)
        const state = {
            error: '',
            songs: null,
            weather: null,
            genreChoice: null,
            id: null,
            artists: null,
            topArtists: null,
            userPlaylists: null,
            playlistId: null,
            snapshot: null,
            targetEnergy: null,
            targetValence: null,
            targetTempo: null,
            targetPopularity: null,
            setError: this.setError,
            clearError: this.clearError,
            setSongs: this.setSongs,
            setWeather: this.setWeather,
            setGenreChoice: this.setGenreChoice,
            setId: this.setId,
            setArtists: this.setArtists,
            setTopArtists: this.setTopArtists,
            setPlaylistId: this.setPlaylistId,
            setUserPlaylists: this.setUserPlaylists,
            setSnapshot: this.setSnapshot,
            setTargetEnergy: this.setTargetEnergy,
            setTargetValence: this.setTargetValence,
            setTargetTempo: this.setTargetTempo,
            setTargetPopularity: this.setTargetPopularity
          }   
        this.state = state; 
    }

    setError = error => {
      console.error(error)
      this.setState({ error })
    }

    clearError = () => {
      this.setState({ error: '' })
    }

    setUserPlaylists = (userPlaylists) => {
      this.setState({ userPlaylists })
    }

    setSongs = (songs) => {
      this.setState({ songs })
    }

    setWeather = (weather) => {
      this.setState({ weather })
    }

    setGenreChoice = (genreChoice) => {
      this.setState({ genreChoice })
    }

    setId = (id) => {
      this.setState({ id })
    }

    setArtists = (artists) => {
      this.setState({ artists })
    }

    setTopArtists = (topArtists) => {
      this.setState({ topArtists })
    }

    setPlaylistId = (playlistId) => {
      this.setState({ playlistId })
    }

    setSnapshot = (snapshot) => {
      this.setState({ snapshot })
    }

    setTargetEnergy = (targetEnergy) => {
      this.setState({ targetEnergy })
    }

    setTargetValence = (targetValence) => {
      this.setState({ targetValence })
    }

    setTargetTempo = (targetTempo) => {
      this.setState({ targetTempo })
    }

    setTargetPopularity = (targetPopularity) => {
      this.setState({ targetPopularity })
    }

    render() {
        return (
            <WindChimeContext.Provider value={this.state}>
                {this.props.children}
            </WindChimeContext.Provider>
        )
    }
}
