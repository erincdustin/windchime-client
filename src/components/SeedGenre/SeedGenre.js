import React from 'react';
import GenreList from '../../GenreList';
import { Link } from 'react-router-dom';
import './SeedGenre.css'

function SeedGenre(props) {

  const mappedGenres = GenreList.map((genre, index) => {
    return(
      <span className="genre-list" key={index}>
        <button 
        value={genre}
        onClick={async (e) =>{
          try {
          console.log(e.target.value);
          await props.setGenre(e.target.value);
          await props.getGenrePlaylist();
          await props.history.push('/results');
          } catch(err) {
            console.log(err)
          }
        }}
        className="btn genre" type="button">{genre}</button>
      </span>)
    });

  return (
    <div>
      <div className="ribbon two"><div className="ribbon-header">Playlist Options</div></div>
      <span className="center">
          <button className="inline btn artist orange" onClick={()=> {
            props.getArtistPlaylist();
            props.history.push('/results');
            }}>Use My Top Artists!</button>
      <h4>OR Pick a genre:</h4>
      <div className="mapped-genres">{mappedGenres}</div>
      {/* <button className="btn-default btn"><Link className="link" to="/getWeather">Back</Link></button> */}
      </span>
    </div>
  );
}

export default SeedGenre;