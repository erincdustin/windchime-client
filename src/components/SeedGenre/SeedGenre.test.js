import React from 'react';
import renderer from 'react-test-renderer';
import { library } from '@fortawesome/fontawesome-svg-core'
import { faWind } from '@fortawesome/free-solid-svg-icons'
import { WindChimeProvider } from '../../contexts/windchime-context';
import SeedGenre from './SeedGenre';

library.add(faWind)

describe(`SeedGenre`, () => {
  it('renders as expected', () => {
    const tree = renderer
      .create(<WindChimeProvider><SeedGenre /></WindChimeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})