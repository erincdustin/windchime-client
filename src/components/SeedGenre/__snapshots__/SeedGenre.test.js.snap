// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeedGenre renders as expected 1`] = `
<div>
  <div
    className="ribbon two"
  >
    <div
      className="ribbon-header"
    >
      Playlist Options
    </div>
  </div>
  <span
    className="center"
  >
    <h4>
      Seed the playlist Using your top Spotify artists:
    </h4>
    <div
      className="top-five"
    />
    <button
      className="inline btn artist orange"
      onClick={[Function]}
    >
      Use My Top Artists
    </button>
    <h4>
      OR Pick from a list of genres:
    </h4>
    <div
      className="mapped-genres"
    >
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="acoustic"
        >
          acoustic
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="alternative"
        >
          alternative
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="ambient"
        >
          ambient
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="bluegrass"
        >
          bluegrass
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="blues"
        >
          blues
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="classical"
        >
          classical
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="country"
        >
          country
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="edm"
        >
          edm
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="electronic"
        >
          electronic
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="folk"
        >
          folk
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="guitar"
        >
          guitar
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="hip-hop"
        >
          hip-hop
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="holidays"
        >
          holidays
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="house"
        >
          house
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="indie"
        >
          indie
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="indie-pop"
        >
          indie-pop
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="jazz"
        >
          jazz
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="latino"
        >
          latino
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="minimal-techno"
        >
          minimal-techno
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="movies"
        >
          movies
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="piano"
        >
          piano
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="pop"
        >
          pop
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="punk-rock"
        >
          punk-rock
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="r-n-b"
        >
          r-n-b
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="road-trip"
        >
          road-trip
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="rock"
        >
          rock
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="romance"
        >
          romance
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="soul"
        >
          soul
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="soundtracks"
        >
          soundtracks
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="study"
        >
          study
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="summer"
        >
          summer
        </button>
      </span>
      <span
        className="genre-list"
      >
        <button
          className="btn genre"
          onClick={[Function]}
          type="button"
          value="techno"
        >
          techno
        </button>
      </span>
    </div>
  </span>
</div>
`;
